### ISSUE TITLE
Dynamic import should accept multiple arguments

### ISSUE THREAD

**Issue by Jack-Works** [tc39/ecma262#1991]
Current syntax:

```
ImportCall[Yield, Await]:
    import(AssignmentExpression[+In, ?Yield, ?Await])
```

Proposed syntax:
```
ImportCall[Yield, Await]:
    import Arguments[+In, ?Yield, ?Await]
```

Currently `import("a", { })` is SyntaxError and it's harder for developers to do "feature detecting" in future. The module attribute proposal is going to add 2nd parameter for `ImportCall` but developers can't write code like this:

```js
try {
    return await import("./a.js", { attribute: something })
} catch(e) {
    return import("./a.js")
}
```

Because it is a SyntaxError. IMO it's better to allow more than 1 parameter and throw a runtime error so it will be easier for the developers in the future.

Since the module attribute is just stage 1, make this change in that proposal will be too late. Too many versions of the browser will only accept the exact 1 argument and it's impossible to write the feature detect above.

It's better to extend this syntax asap for the future (and throw at runtime for now) and it won't break anything (because it just makes invalid syntax valid).

**Comment by WebReflection** [comment]
so, basically you would prefer this?

```js
return await import("./a.js", { attribute: something }).catch(() => import("./a.js"));
```

it makes it less obvious that the `import(...)` supports or not arguments (the feature detection you mentioned) but since detecting dynamic `import.length` is also not an option, I guess it *might* make sense.

**Comment by Jack-Works** [comment]
`import.length` is an interesting idea but it doesn't resolve the problem:

```js
// if there is a import.length meta property
if (import.length > 2) import("a", {}) // oops, still SyntaxError. Can not run
else import("a")
```

**Comment by littledan** [comment]
I don't think this makes sense to add separately from the module attributes proposal. Given that there are browsers shipping now without this syntax change, you'd need your feature test to work there too. Let's just continue discussion in the module attributes proposal.

**Comment by Jack-Works** [comment]
This can be changed earlier than that proposal so in future it is less pain. (Less range of versions of browsers in this embarrassing situation)

**Comment by littledan** [comment]
I don't think I agree that this is important to be able to feature test for. It is really unclear to me how/whether this would be usable in practice, even if it were supported. The typical usage pattern is with a static import, where it will be a syntax error in browsers without module attributes. I'd recommend people initially adopt module attributes similarly to async/await: through transpilers.

Can we move this discussion to the module attributes repo?

**Comment by Jack-Works** [comment]
You can move it
